lib/pengine/unpack.c

```diff
@@ -2675,9 +2670,13 @@ unpack_rsc_op_failure(resource_t * rsc, node_t * node, int rc, xmlNode * xml_op,
             rsc->role = RSC_ROLE_STOPPED;
 
         } else {
-            crm_warn("Forcing %s to stop after a failed demote action", rsc->id);
+            /*
+             * Staying in master role would put the PE/TE into a loop. Setting
+             * slave role is not dangerous because the resource will be stopped
+             * as part of recovery, and any master promotion will be ordered
+             * after that stop.
+             */
             rsc->role = RSC_ROLE_SLAVE;
-            rsc->next_role = RSC_ROLE_STOPPED;
```

Comment:
Doesn't seem like the comment matches the code anymore (since we're not forcing it to stop)

Reply:
AFAICT we are still forcing it to stop, we're just not forcing it to stay stopped.

Getting into this block means we have a failed demote action. When we get to native_create_actions(), if next role is not already stopped, native_create_actions() will force recovery due to there being a current node and a chosen node -- so we get a stop and a start.